src/VirtualPage.php

```diff
@@ -146,10 +146,21 @@ public function getHTML()
                 }
             }
 
+            // Prepare modals in order to include them in VirtualPage.
+            $modalHtml = '';
+            foreach ($this->app->html !== null ? $this->app->html->elements : [] as $view) {
+                if ($view instanceof Modal) {
+                    $modalHtml .= $view->getHTML();
+                    $this->app->layout->_js_actions = array_merge($this->app->layout->_js_actions, $view->_js_actions);
+                }
+            }
+
             $this->app->layout->template->setHTML('Content', parent::getHTML());
             $this->app->layout->_js_actions = array_merge($this->app->layout->_js_actions, $this->_js_actions);
 
             $this->app->html->template->setHTML('Content', $this->app->layout->getHTML());
+            $this->app->html->template->setHtml('Modals', $modalHtml);
```

Comment:
setHTML or setHtml? when using strict_types this will fail if case does not match afaik.

Reply:
Thanks for catching. Updated!